File: core-functionality/src/main/java/core/process/api/http/BusinessProcessController.java
package core.process.api.http;

import core.process.api.services.BusinessProcessService;
import domain.process.BusinessProcess;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.UUID;

@RestController
@RequestMapping("/api/v1/processes")
public class BusinessProcessController {

    @Autowired
    private BusinessProcessService businessProcessService;

    @PostMapping
    public ResponseEntity<BusinessProcess> createBusinessProcess(@RequestBody BusinessProcess businessProcess) {
        BusinessProcess createdPattern = businessProcessService.createBusinessProcess(businessProcess);
        return ResponseEntity.ok(createdPattern);
    }

    @GetMapping
    public ResponseEntity<List<BusinessProcess>> getAllBusinessProcesses() {
        List<BusinessProcess> processes = businessProcessService.getAllBusinessProcesses();
        return ResponseEntity.ok(processes);
    }

    @GetMapping("/id")
    public ResponseEntity<BusinessProcess> getBusinessProcessById(@PathVariable UUID id) {
        BusinessProcess processe = businessProcessService.getBusinessProcess(id);
        return ResponseEntity.ok(processe);
    }

}

File: core-functionality/src/main/kotlin/core/process/api/http/BusinessProcessController.kt
package core.process.api.http

import core.process.domain.BusinessProcessService
import org.springframework.http.HttpStatus
import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/v1/processes")
class BusinessProcessController(
    private val businessProcessService: BusinessProcessService
) {

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    fun createGroup(
        @RequestBody @Validated request: CreateBusinessProcessRequest,
    ): BusinessProcessResponse = businessProcessService.createBusinessProcess(request).toResponse()

}